src/texture.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
```

Comment:
missing license headers, and as it's purely client side cpp/header please move them to client folder